### DISCUSSION TITLE
Ways to fix "LinAlgWarning: Ill-conditioned matrix (rcond=1.29507e-30): result may not be accurate" when

### DISCUSSION THREAD

**Discussion by TiMo3654** [python-control/python-control#858] [Q&A]
I am trying to design a PIDT2 controller for a plant which roughly behaves like a PT2 plant. The control package works really superb for this and I am receiving good results as can be seen in the last bode plot in my repository on [Github](https://github.com/TiMo3654/Bucky/blob/main/buck_controller.ipynb).

I would like to conclude the design with a step response of the open loop transfer function, but I receive the error stated in the title. Are there any ways to prevent this error and to get the correct result?

**Comment by ilayn** [comment]
Could you try again by just using state space models and no transfer functions? This is a warning coming from a line that solves an equation `AX = B` for `X` where `A` gets very close to being singular. In state methods, I don't think there is any such equations but in transfer function branches there might be a few. 

Otherwise if everything looks OK just silence the warning 

```python
import warnings

# your regular code
with warnings.catch_warnings():
    warnings.simplefilter('ignore')
    # codeline that produces a warning
    _, _, _ = step_response(_, _, ...)

```

**Reply by TiMo3654** [reply]
Thanks for the reply! I think my expecations were wrong! By utilizing the feedback function I am able to get the desired step response.
